src/Design.cc

```diff
@@ -208,6 +208,35 @@ bool Design::isSequential(odb::dbMaster* master)
   return lib_cell->hasSequentials();
 }
 
+bool Design::isInClock(odb::dbInst* inst) {
+  for (auto* iterm : inst->getITerms()) {
+    auto* net = iterm->getNet();
+    if (net != NULL && net->getSigType() == odb::dbSigType::CLOCK){
```

Comment:
Note that this will only work post-CTS.  If you want it to work earlier you'll have to query sta directly.